### DISCUSSION TITLE
How does roles work?

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#1787] [Q&A]
How does roles work?

For example, I've created role field with option (super-admin, manager) in users collection.
What I want is that only super-admin can retrieve the fulll list and the specific users without the admin role only their row.

![image](https://user-images.githubusercontent.com/31880091/216845210-bcad4bd9-4c56-4a2e-8b86-aca62cb0ec65.png)

I tried to use ````@request.auth.role = 'super-admin'```` but the list is still empty

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the question. What is `role` - single or multiple `select` field?
If it a single (aka. "Max Select = 1") then you can search directly by comparing the value as above.
If it is multiple, then you can either use the `:each` modifier to apply a condition on each selection or the `~` for a "contain/like" operation, eg. `@request.auth.role ~ '"super-admin"'`.
